src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java

```diff
@@ -115,9 +115,9 @@
  * The search order of locale sensitive services can
  * be configured by using the {@systemProperty java.locale.providers} system property.
  * This system property declares the user's preferred order for looking up
- * the locale sensitive services separated by a comma. It is only read at
- * the Java runtime startup, so the later call to System.setProperty() won't
- * affect the order.
+ * the locale sensitive services separated by a comma. It is only read and cached at
+ * the initialization of this class, so the later call to
+ * {@link System#setProperty(String, String)} may not affect the order.
```

Comment:
I wonder if we can be clearer as "may not" implies uncertainty. While it indeed may or may not work due to the timing of the initialization of this class, my understanding of the above statement is that it implied the runtime startup is recommended as it provides assurance. Would it be better to put that in the statement? sth. like: It is read once and cached at the Java runtime startup or initialization of this class. A call after the initialization of this class will not affect the order.

There's still uncertainty

Reply:
It was intentional to use `may not` because as you said, there's still uncertainty. To clarify it more, I added wording that `setProperty` use is discouraged to change the preferred order.

Reply:
That sounds good to me, suggesting doing it at launch while discouraging the use of setProperty.

Reply:
Hi Lance,

> * Are there any scenarios where invoking setProperty will not override the command line setting ?

Yes. For example, if a logger is installed, it formats the log date/time which causes initialization of locale-related classes before the app's main() method invocation, resulting setProperty() having no effect.

> * Did you consider an `@ImplNote` for your clarification given the behavior "might" be different in other implementations (I am not sure myself) and is implementation defined?

As in the above example, the behavior changes regardless of the implementation of the runtime. The clarification is mainly for the API consumer, rather than platform implementors.

Hope this answers your questions.

Reply:
Hi Naoto,

Thank you for the clarification.  Yes it makes sense.

Best
Lance